src/OpenTelemetry/Internal/SelfDiagnosticsConfigRefresher.cs

```diff
@@ -44,12 +45,24 @@ internal class SelfDiagnosticsConfigRefresher : IDisposable
 
     public SelfDiagnosticsConfigRefresher()
     {
-        this.configParser = new SelfDiagnosticsConfigParser();
-        this.UpdateMemoryMappedFileFromConfiguration();
+        if (this.IsSelfDiagnosticsEnVarOn)
```

Comment:
What should happen if the environment variable is not set during app startup?

Reply:
Another option is to add a property to the config file, allowing us to manage all the configurations in one place.

Reply:
@rajkumar-rangaraj The purpose of this PR is for scenario where user cannot access files on disk.

Initially my concern was that the environment variables wouldn't be picked up if they're not set during construction, however, on second thought, env vars cannot be set after a program is started anyways, so initializing only during construction isn't an issue.

This comment can be resolved unless anyone has other concerns.